File: cp-trials/src/main/java/in/shabhushan/cp_trials/DisemvowelTrolls.java
package in.shabhushan.cp_trials;

import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class DisemvowelTrolls {
    private static List<Character> vowels = Arrays.asList('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U');

    public static String disemvowel(String str) {
        return str.chars()
                .mapToObj(i -> (char)i)
                .filter(it -> !vowels.contains(it))
                .map(String::valueOf)
                .collect(Collectors.joining());
    }
}

File: cp-trials/src/main/kotlin/in/shabhushan/cp_trials/DisemvowelTrolls.kt
package `in`.shabhushan.cp_trials

val vowels: List<Char> = listOf('a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U')

fun disemvowel(str: String): String {
    return str.toCharArray()
//        .partition {
//            !vowels.contains(it)
//        }.let {
//            it.first.joinToString("")
//        }
        .filter { !vowels.contains(it)}
        .joinToString("")

}
fun main() {
    println(disemvowel("This website is for losers LOL!"))
}
